src/Features/Core/Portable/Diagnostics/EngineV2/DiagnosticIncrementalAnalyzer.StateManager.cs

```diff
@@ -308,6 +309,27 @@ private void VerifyProjectDiagnosticStates(IEnumerable<StateSet> stateSets)
 
                 VerifyUniqueStateNames(hostStates.Concat(stateSets));
             }
+
+            private readonly struct HostAnalyzerStateSetKey : IEquatable<HostAnalyzerStateSetKey>
+            {
+                public HostAnalyzerStateSetKey(string language, HostDiagnosticAnalyzers analyzers)
+                {
+                    Language = language;
+                    Analyzers = analyzers;
+                }
+
+                public string Language { get; }
+                public HostDiagnosticAnalyzers Analyzers { get; }
```

Comment:
📝 I requested that instead of using `HostDiagnosticAnalyzers` as the key directly, we use `HostDiagnosticAnalyzers.GetHostAnalyzerReferencesMap()`. This has the same net result for the keys but ensures we don't accidentally GC root the `DiagnosticAnalyzer` instances themselves (from other fields in `HostDiagnosticAnalyzers`) in the off chance this cache ends up with unforeseen items.